### DISCUSSION TITLE
docker alist从3.10更新到3.11的问题

### DISCUSSION THREAD

**Discussion by yuhouyanran** [AlistGo/alist#3432] [Q&A] [answered]
首先我我使用了停止容器--删除旧镜像--运行拉取最新版alist镜像命令--建立新的alist容器并配置参数。
相关命令如下：
`docker stop alist`
`docker rm alist`
`docker pull xhofe/alist:latest`
`docker run -d --restart=always -v /etc/alist:/opt/alist/data -p 5244:5244 --name="alist" xhofe/alist:latest`
之后重新打开alist页面查看版本依然是3.10，且挂载的两个存储还在，遂以为有可能是没有重启软路由所致，所以又进行了以下一次步骤依：停止容器--删除旧镜像--重启软路由--运行拉取最新版alist镜像命令--建立新的alist容器并配置参数，再次进入网页查看版本依然是3.10且有两个已成功挂载的存储。
而后尝试了建立watchower来自动更新docker下所有容器并手动运行了一次watchower，显示以下代码
<a href="https://imgse.com/i/pS7dp2q"><img src="https://s1.ax1x.com/2023/02/15/pS7dp2q.png" alt="pS7dp2q.png" border="0" /></a>

**Comment by yuhouyanran** [comment, marked as answer]
请问如何解决，蟹蟹

**Reply by hangyubin** [reply]
拉取镜像时指定3.11.0版

**Comment by 360278871** [comment]
额。。我好像拉不到

**Reply by hangyubin** [reply]
docker pull xhofe/alist:3.11.0

**Reply by yuhouyanran** [reply]
OK了，但是你这代码有点问题2333，我跑了一遍不行，然后看到3.11.0前面没有v我就自己加了个v试了一下拉取到了，部署容器的时候也把最后面的latest改成v3.11.0了，打开网页显示v3.11.0是最新版了。还有就是想问一下为啥会出现这种问题，明明3.11.0不就是最新版吗，怎么用latest拉取就一直是老版本3.10.1，很奇怪，非得固定成3.11.0才行，拉取更新其他容器也没这问题啊，第一次碰到，能不能别用术语解释，说人话2333

**Reply by hangyubin** [reply]
这个原因我也不知道！拉了几次都是3.10.1，后在这个网站看了就指定版本搞定

**Reply by hangyubin** [reply]
xhofe/alist，这应是作者维护的镜像 
https://hub.docker.com/r/xhofe/alist/tags

**Reply by yuhouyanran** [reply]
奥，蟹蟹

**Reply by JohnDoeCN** [reply]
遇到过一样问题，貌似你拉取过一次latest之后，再拉取latest就锁定了第一次拉取的版本，原因未知。
